contracts/cw1-subkeys/src/contract.rs

```diff
@@ -88,6 +90,44 @@ where
                     allowance.balance = allowance.balance.sub(amount.clone())?;
                     allowances.save(owner_raw.as_slice(), &allowance)?;
                 }
+                CosmosMsg::Staking(StakingMsg::Delegate {
```

Comment:
This makes a quite large branch.
I would do something like:

`CosmosMsg::Staking(staking_msg) => check_staking_message(&staking_msg, &mut allowance)?` 

or such and then include all these branches inside `check_staking_message`. That function would then be simpler:

```rust
match staking_msg {
  StakingMsg:::Delegate { amount, ..} => 
                    if !allowance.permissions.delegate {
                        Err(StdError::generic_err(PermissionErr::Delegate{}))
                        // or maybe this (and cast to StdError above when `?` is used)
                        Err(PermissionErr::Delegate{})
                    } else {
                       Ok(())
                   }
}
```

Try to pull some out and simplify